win32/src/gballoc_ll_jemalloc.c

```diff
@@ -214,3 +218,240 @@ void gballoc_ll_print_stats(void)
     /* Codes_SRS_GBALLOC_LL_JEMALLOC_01_008: [ gballoc_ll_print_stats shall call je_malloc_stats_print and pass to it jemalloc_print_stats_callback as print callback. ]*/
     je_malloc_stats_print(jemalloc_print_stats_callback, NULL, NULL);
 }
+
+static int gballoc_ll_set_dirty_decay(int64_t decay_milliseconds)
+{
+    int result;
+
+    if (decay_milliseconds < -1)
+    {
+        /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_019: [ If decay_milliseconds is less than -1, gballoc_ll_set_option shall fail and return a non-zero value. ]*/
+        LogError("decay_milliseconds must be -1 or greater, decay_milliseconds=%" PRId64 "", decay_milliseconds);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        int64_t old_decay_milliseconds;
+        size_t old_decay_milliseconds_size = sizeof(old_decay_milliseconds);
+        MU_STATIC_ASSERT(sizeof(int64_t) == sizeof(size_t));
+
+        /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_005: [ gballoc_ll_set_option shall retrieve the old dirty decay value and set the new dirty decay value to decay_milliseconds for new arenas by calling je_mallctl with arenas.dirty_decay_ms as the command. ]*/
+        if (je_mallctl("arenas.dirty_decay_ms", &old_decay_milliseconds, &old_decay_milliseconds_size, &decay_milliseconds, sizeof(decay_milliseconds)) != 0)
+        {
+            /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_018: [ If there are any errors, gballoc_ll_set_option shall fail and return a non-zero value. ]*/
+            LogError("je_mallctl(const char* name=arenas.dirty_decay_ms, void* oldp=%p, size_t* oldlenp=%p, void* newp=%p, size_t newlen=%zu) failed", &old_decay_milliseconds, &old_decay_milliseconds_size, &decay_milliseconds, sizeof(decay_milliseconds));
```

Comment:
>const char* [](http://example.com/codeflow?start=33&length=11)

no need to specify the parameter types when logging an error from a called function.

Here and everywhere

Reply:
Done